apps/ledger-live-mobile/src/reducers/accounts.ts

```diff
@@ -478,12 +479,25 @@ export const orderedNftsSelector = createSelector(
  * ```
  * */
 export const orderedVisibleNftsSelector = createSelector(
-  orderedNftsSelector,
-  hiddenNftCollectionsSelector,
-  (nfts, hiddenNftCollections) =>
-    nfts.filter(
+  accountsSelector,
+  nftCollectionsStatusByNetworkSelector,
+  (_: State, hideSpam: boolean) => hideSpam,
+  (accounts, nftCollectionsStatusByNetwork, hideSpam) => {
+    const nfts = accounts.map(a => a.nfts ?? []).flat();
+
+    const hiddenNftCollections = Object.values(nftCollectionsStatusByNetwork).flatMap(contracts =>
+      Object.entries(contracts)
```

Comment:
(this is just a none required note), i think you would rather use the parser used here apps/ledger-live-mobile/src/hooks/nfts/useNftCollectionsStatus.ts. But i understand the structure fact of passing by a custom hook doesn't make sens to use it in a reducer. None blocking for me, but i will affect code duplication score.